pkg/model/components/containerd.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"fmt"
+
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/loader"
+)
+
+// ContainerdOptionsBuilder adds options for containerd to the model
+type ContainerdOptionsBuilder struct {
+	*OptionsContext
+}
+
+var _ loader.OptionsBuilder = &ContainerdOptionsBuilder{}
+
+// BuildOptions is responsible for filling in the default setting for containerd daemon
+func (b *ContainerdOptionsBuilder) BuildOptions(o interface{}) error {
+	clusterSpec := o.(*kops.ClusterSpec)
+
+	if clusterSpec.Containerd == nil {
+		clusterSpec.Containerd = &kops.ContainerdConfig{}
+	}
+
+	containerd := clusterSpec.Containerd
+
+	if fi.StringValue(clusterSpec.Containerd.Version) == "" {
+		containerdVersion := ""
+
+		if clusterSpec.ContainerRuntime == "containerd" {
+			if clusterSpec.KubernetesVersion == "" {
+				return fmt.Errorf("Kubernetes version is required")
+			}
+
+			sv, err := KubernetesVersion(clusterSpec)
```

Comment:
Tip (for the future): We do have the `IsKubernetesGTE` helper in a bunch of places, not sure if it's available here.

Reply:
I tried to keep this as close as possible to the Docker version to make it easier to review.
The IsKubernetesGTE function family is quite nice. Thanks.